modules/core/src/test/java/org/apache/ignite/internal/processors/failure/FailureProcessorThreadDumpThrottlingTest.java

```diff
@@ -0,0 +1,285 @@
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Predicate;
+
+import com.google.common.collect.ImmutableSet;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.IgniteSystemProperties;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.failure.FailureContext;
+import org.apache.ignite.failure.FailureType;
+import org.apache.ignite.failure.TestFailureHandler;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.apache.ignite.testframework.ListeningTestLogger;
+import org.apache.ignite.testframework.junits.WithSystemProperty;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.jetbrains.annotations.Nullable;
+import org.junit.Test;
+
+import static org.apache.ignite.failure.FailureType.SYSTEM_CRITICAL_OPERATION_TIMEOUT;
+import static org.apache.ignite.failure.FailureType.SYSTEM_WORKER_BLOCKED;
+import static org.apache.ignite.internal.util.IgniteUtils.THREAD_DUMP_MSG;
+
+/**
+ * Tests for throttling thread dumps during handling failures.
+ */
+@WithSystemProperty(key = IgniteSystemProperties.IGNITE_DUMP_THREADS_ON_FAILURE, value = "true")
+public class FailureProcessorThreadDumpThrottlingTest extends GridCommonAbstractTest {
+    /** Test logger. */
+    private static TestLogger testLog;
+
+    /** {@inheritDoc} */
+    @Override protected IgniteConfiguration getConfiguration(String igniteInstanceName) throws Exception {
+        IgniteConfiguration cfg =  super.getConfiguration(igniteInstanceName);
+
+        TestFailureHandler hnd = new TestFailureHandler(false);
+
+        hnd.setIgnoredFailureTypes(ImmutableSet.of(FailureType.SYSTEM_CRITICAL_OPERATION_TIMEOUT, SYSTEM_WORKER_BLOCKED));
+
+        cfg.setFailureHandler(hnd);
+
+        cfg.setGridLogger(testLog);
+
+        return cfg;
+    }
+
+    /**
+     * Tests that thread dumps will not get if {@code IGNITE_DUMP_THREADS_ON_FAILURE == false}.
+     */
+    @Test
+    @WithSystemProperty(key = IgniteSystemProperties.IGNITE_DUMP_THREADS_ON_FAILURE, value = "false")
+    @WithSystemProperty(key = IgniteSystemProperties.IGNITE_DUMP_THREADS_ON_FAILURE_THROTTLING_TIMEOUT, value = "0")
+    public void testNoThreadDumps() throws Exception {
+        IgniteEx ignite = null;
+
+        try {
+            testLog = new TestLogger(log, Collections.singleton(s -> s.contains(THREAD_DUMP_MSG)));
+
+            ignite = (IgniteEx)startGrid();
+
+            FailureContext failureCtx =
+                    new FailureContext(SYSTEM_WORKER_BLOCKED, new Throwable("Failure context error"));
+
+            for (int i = 0; i < 3; i++)
+                ignite.context().failure().process(failureCtx);
+
+            assertTrue(testLog.filteredRecords().isEmpty());
+        }
+        finally {
```

Comment:
Is this now preferred to using @Before/@After or beforeTest/afterTest methods?

Reply:
It seems there is no rule for it at present. It is easier to find method by names than by annotations.